corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py

```diff
@@ -294,3 +302,191 @@ def test_conflict(self):
 
         # attempt to create the same doc twice shouldn't fail
         self._send_to_es_and_check(doc)
+
+
+@skipIf(settings.ELASTICSEARCH_MAJOR_VERSION != 7, 'Only applicable for ES7')
+@es_test
+class TestILM(SimpleTestCase):
+
+    def setUp(self):
+        self.es = get_es_new()
+        self.es_interface = ElasticsearchInterface(self.es)
+        self.index = XFORM_INDEX_INFO.index
+        self.alias = XFORM_INDEX_INFO.alias
+        XFORM_INDEX_INFO.ilm_config = prefix_for_tests(MAX_DOCS)
+        self.es.cluster.put_settings({
+            "persistent": {"indices.lifecycle.poll_interval": "1s"}
+        })
+
+        with trap_extra_setup(ConnectionError):
+            ensure_index_deleted(self.index)
+            initialize_index_and_mapping(self.es, XFORM_INDEX_INFO)
+
+    def tearDown(self):
+        XFORM_INDEX_INFO.ilm_config = ""
+        ensure_index_deleted(self.index)
+
+    def rollover(self):
+        # wait for ILM to kick in.
+        #   Is double the poll_interval to provide enough buffer for ILM process
+        time.sleep(2)
+
+    def _send_to_es(self, docs):
+        for chunk in chunked(docs, 2):
+            # 2 is the max_docs per the policy
+            for doc in chunk:
+                send_to_es('forms', doc)
+                self.es.indices.refresh(self.alias)
+            self.rollover()
+
+    def test_index_rollsover(self):
+        #index should rollover 2 times making 3 total indices
+        self._send_to_es([
+            {"_id": "d1", "prop": "a"},
+            {"_id": "d2", "prop": "b"},
+            {"_id": "d3", "prop": "c"},
+            {"_id": "d4", "prop": "d"},
+            {"_id": "d5", "prop": "e"},
+        ])
+        self.assertEqual(
+            len(get_indices_by_alias(self.alias)),
+            3
+        )
+        self.assertEqual(
+            FormES().remove_default_filters().ids_query(['d1', 'd4']).exclude_source().run().doc_ids,
+            ['d1', 'd4']
```

Comment:
what is this testing? One possible improvement I can think of would be to inspect the raw results to see which indexes the docs came from.

Reply:
Sure, will fix it